agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -394,6 +394,45 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
     return true;
 }
 
+bool slave_thread::start_monitoring(const beerocks_message::sClientMonitoringParams &params,
```

Comment:
It is great that you split the start monitoring to a separate function. However, this should be done in a separate commit.